### ISSUE TITLE
Hangs on ansible provisionning 

### ISSUE THREAD

**Issue by quanticware** [hashicorp/packer#8854]
#### Overview of the Issue

I begin in packer world. 
When I launch packer with ansible provisionning, it stucks on first task. I tried to gather_facts no, with root user etc. 
It's stuck.

#### Reproduction Steps

Build packer image

### Packer version

1.5.4

### Ansible version:

2.9.2

### Simplified Packer Buildfile

```
{
    "variables": {
    },
    "builders": [
        {
          "type": "lxd",
          "name": "ubuntu-1804-qw",
          "image": "ubuntu-minimal:bionic",
          "output_image": "ubuntu-1804-qw",
          "publish_properties": {
            "description": "Image LXD ubuntu bionic ansible ready"
          }
      }
    ],
    "provisioners": [
      {
        "type": "shell",
        "script": "../play/ansible_user_setup.sh"
      },
      {
        "type": "ansible",
        "user": "ansible",
        "playbook_file": "config.yml",
        "extra_arguments": ["-vvvv"]
      }
    ]
}
```
### Operating system and Environment details

Ubuntu 18.04.2

### Log Fragments and crash.log files

```

2020/03/09 01:03:20 [INFO] Packer version: 1.5.4 [go1.13.7 linux amd64]
2020/03/09 01:03:20 Checking 'PACKER_CONFIG' for a config file path
2020/03/09 01:03:20 'PACKER_CONFIG' not set; checking the default config file path
2020/03/09 01:03:20 Attempting to open config file: /home/fred/.packerconfig
2020/03/09 01:03:20 [WARN] Config file doesn't exist: /home/fred/.packerconfig
2020/03/09 01:03:20 Setting cache directory: /labo/infrabuilder/packer/packer_cache
2020/03/09 01:03:20 Creating plugin client for path: /usr/local/bin/packer
2020/03/09 01:03:20 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-builder-lxd"}
2020/03/09 01:03:20 Waiting for RPC address for: /usr/local/bin/packer
2020/03/09 01:03:20 packer-builder-lxd plugin: [INFO] Packer version: 1.5.4 [go1.13.7 linux amd64]
2020/03/09 01:03:20 packer-builder-lxd plugin: Checking 'PACKER_CONFIG' for a config file path
2020/03/09 01:03:20 packer-builder-lxd plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/03/09 01:03:20 Received unix RPC address for /usr/local/bin/packer: addr is /tmp/packer-plugin779711394
2020/03/09 01:03:20 packer-builder-lxd plugin: Attempting to open config file: /home/fred/.packerconfig
2020/03/09 01:03:20 packer-builder-lxd plugin: [WARN] Config file doesn't exist: /home/fred/.packerconfig
2020/03/09 01:03:20 packer-builder-lxd plugin: Setting cache directory: /labo/infrabuilder/packer/packer_cache
2020/03/09 01:03:20 packer-builder-lxd plugin: args: []string{"packer-builder-lxd"}
2020/03/09 01:03:20 packer-builder-lxd plugin: Plugin address: unix /tmp/packer-plugin779711394
2020/03/09 01:03:20 packer-builder-lxd plugin: Waiting for connection...
2020/03/09 01:03:20 packer-builder-lxd plugin: Serving a plugin connection...
2020/03/09 01:03:20 Creating plugin client for path: /usr/local/bin/packer
2020/03/09 01:03:20 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-provisioner-shell"}
2020/03/09 01:03:20 Waiting for RPC address for: /usr/local/bin/packer
2020/03/09 01:03:20 packer-provisioner-shell plugin: [INFO] Packer version: 1.5.4 [go1.13.7 linux amd64]
2020/03/09 01:03:20 packer-provisioner-shell plugin: Checking 'PACKER_CONFIG' for a config file path
2020/03/09 01:03:20 packer-provisioner-shell plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/03/09 01:03:20 packer-provisioner-shell plugin: Attempting to open config file: /home/fred/.packerconfig
2020/03/09 01:03:20 packer-provisioner-shell plugin: [WARN] Config file doesn't exist: /home/fred/.packerconfig
2020/03/09 01:03:20 packer-provisioner-shell plugin: Setting cache directory: /labo/infrabuilder/packer/packer_cache
2020/03/09 01:03:20 packer-provisioner-shell plugin: args: []string{"packer-provisioner-shell"}
2020/03/09 01:03:20 packer-provisioner-shell plugin: Plugin address: unix /tmp/packer-plugin175738786
2020/03/09 01:03:20 packer-provisioner-shell plugin: Waiting for connection...
2020/03/09 01:03:20 Received unix RPC address for /usr/local/bin/packer: addr is /tmp/packer-plugin175738786
2020/03/09 01:03:20 packer-provisioner-shell plugin: Serving a plugin connection...
2020/03/09 01:03:20 Creating plugin client for path: /usr/local/bin/packer
2020/03/09 01:03:20 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-provisioner-ansible"}
2020/03/09 01:03:20 Waiting for RPC address for: /usr/local/bin/packer
2020/03/09 01:03:20 packer-provisioner-ansible plugin: [INFO] Packer version: 1.5.4 [go1.13.7 linux amd64]
2020/03/09 01:03:20 packer-provisioner-ansible plugin: Checking 'PACKER_CONFIG' for a config file path
2020/03/09 01:03:20 packer-provisioner-ansible plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/03/09 01:03:20 packer-provisioner-ansible plugin: Attempting to open config file: /home/fred/.packerconfig
2020/03/09 01:03:20 packer-provisioner-ansible plugin: [WARN] Config file doesn't exist: /home/fred/.packerconfig
2020/03/09 01:03:20 packer-provisioner-ansible plugin: Setting cache directory: /labo/infrabuilder/packer/packer_cache
2020/03/09 01:03:20 packer-provisioner-ansible plugin: args: []string{"packer-provisioner-ansible"}
2020/03/09 01:03:20 packer-provisioner-ansible plugin: Plugin address: unix /tmp/packer-plugin098603982
2020/03/09 01:03:20 packer-provisioner-ansible plugin: Waiting for connection...
2020/03/09 01:03:20 Received unix RPC address for /usr/local/bin/packer: addr is /tmp/packer-plugin098603982
2020/03/09 01:03:20 packer-provisioner-ansible plugin: Serving a plugin connection...
2020/03/09 01:03:20 Build debug mode: false
2020/03/09 01:03:20 Force build: false
2020/03/09 01:03:20 On error: 
2020/03/09 01:03:20 Preparing build: ubuntu-1804-qw
ubuntu-1804-qw: output will be in this color.

2020/03/09 01:03:21 packer-provisioner-ansible plugin: ansible-playbook version: 2.9.6
2020/03/09 01:03:21 Waiting on builds to complete...
2020/03/09 01:03:21 Starting build run: ubuntu-1804-qw
2020/03/09 01:03:21 Running builder: lxd
2020/03/09 01:03:21 [INFO] (telemetry) Starting builder lxd
==> ubuntu-1804-qw: Creating container...
2020/03/09 01:03:21 packer-builder-lxd plugin: Executing lxc command: []string{"launch", "--ephemeral=false", "--profile=default", "ubuntu-minimal:bionic", "packer-ubuntu-1804-qw"}
2020/03/09 01:03:31 packer-builder-lxd plugin: stdout: Creating packer-ubuntu-1804-qw
2020/03/09 01:03:31 packer-builder-lxd plugin: Starting packer-ubuntu-1804-qw
2020/03/09 01:03:31 packer-builder-lxd plugin: stderr:
2020/03/09 01:03:34 packer-builder-lxd plugin: Sleeping for 3 seconds...
2020/03/09 01:03:34 packer-builder-lxd plugin: Unable to load communicator config from state to populate provisionHookData
2020/03/09 01:03:34 packer-builder-lxd plugin: Running the provision hook
2020/03/09 01:03:34 [INFO] (telemetry) Starting provisioner shell
==> ubuntu-1804-qw: Provisioning with shell script: ../play/ansible_user_setup.sh
2020/03/09 01:03:34 packer-provisioner-shell plugin: Opening ../play/ansible_user_setup.sh for reading
2020/03/09 01:03:34 packer-provisioner-shell plugin: [INFO] 693 bytes written for 'uploadData'
2020/03/09 01:03:34 [INFO] 693 bytes written for 'uploadData'
2020/03/09 01:03:34 packer-builder-lxd plugin: Executing with lxc exec in container: packer-ubuntu-1804-qw test -d "/tmp/script_3927.sh"
2020/03/09 01:03:34 packer-builder-lxd plugin: Executing lxc exec: /bin/sh []string{"/bin/sh", "-c", "lxc exec packer-ubuntu-1804-qw -- /bin/sh -c \"test -d \"/tmp/script_3927.sh\"\""}
2020/03/09 01:03:44 packer-builder-lxd plugin: lxc exec execution exited with '1': 'test -d "/tmp/script_3927.sh"'
2020/03/09 01:03:44 packer-builder-lxd plugin: Running copy command: lxc file push - packer-ubuntu-1804-qw/tmp/script_3927.sh
2020/03/09 01:03:46 packer-builder-lxd plugin: Executing with lxc exec in container: packer-ubuntu-1804-qw chmod 0755 /tmp/script_3927.sh
2020/03/09 01:03:46 packer-builder-lxd plugin: Executing lxc exec: /bin/sh []string{"/bin/sh", "-c", "lxc exec packer-ubuntu-1804-qw -- /bin/sh -c \"chmod 0755 /tmp/script_3927.sh\""}
2020/03/09 01:03:55 packer-builder-lxd plugin: lxc exec execution exited with '0': 'chmod 0755 /tmp/script_3927.sh'
2020/03/09 01:03:55 packer-builder-lxd plugin: [INFO] RPC endpoint: Communicator ended with: 0
2020/03/09 01:03:55 [INFO] RPC client: Communicator ended with: 0
2020/03/09 01:03:55 [INFO] RPC endpoint: Communicator ended with: 0
2020/03/09 01:03:55 packer-provisioner-shell plugin: [INFO] RPC client: Communicator ended with: 0
2020/03/09 01:03:55 packer-builder-lxd plugin: Executing with lxc exec in container: packer-ubuntu-1804-qw chmod +x /tmp/script_3927.sh; PACKER_BUILDER_TYPE='lxd' PACKER_BUILD_NAME='ubuntu-1804-qw'  /tmp/script_3927.sh
2020/03/09 01:03:55 packer-builder-lxd plugin: Executing lxc exec: /bin/sh []string{"/bin/sh", "-c", "lxc exec packer-ubuntu-1804-qw -- /bin/sh -c \"chmod +x /tmp/script_3927.sh; PACKER_BUILDER_TYPE='lxd' PACKER_BUILD_NAME='ubuntu-1804-qw'  /tmp/script_3927.sh\""}
2020/03/09 01:04:05 packer-builder-lxd plugin: lxc exec execution exited with '0': 'chmod +x /tmp/script_3927.sh; PACKER_BUILDER_TYPE='lxd' PACKER_BUILD_NAME='ubuntu-1804-qw'  /tmp/script_3927.sh'
2020/03/09 01:04:05 packer-builder-lxd plugin: [INFO] RPC endpoint: Communicator ended with: 0
2020/03/09 01:04:05 [INFO] 0 bytes written for 'stdout'
2020/03/09 01:04:05 [INFO] 0 bytes written for 'stderr'
2020/03/09 01:04:05 [INFO] RPC client: Communicator ended with: 0
2020/03/09 01:04:05 [INFO] RPC endpoint: Communicator ended with: 0
2020/03/09 01:04:05 packer-provisioner-shell plugin: [INFO] 0 bytes written for 'stdout'
2020/03/09 01:04:05 packer-provisioner-shell plugin: [INFO] 0 bytes written for 'stderr'
2020/03/09 01:04:05 packer-provisioner-shell plugin: [INFO] RPC client: Communicator ended with: 0
2020/03/09 01:04:05 packer-builder-lxd plugin: Executing with lxc exec in container: packer-ubuntu-1804-qw rm -f /tmp/script_3927.sh
2020/03/09 01:04:05 packer-builder-lxd plugin: Executing lxc exec: /bin/sh []string{"/bin/sh", "-c", "lxc exec packer-ubuntu-1804-qw -- /bin/sh -c \"rm -f /tmp/script_3927.sh\""}
2020/03/09 01:04:14 packer-builder-lxd plugin: lxc exec execution exited with '0': 'rm -f /tmp/script_3927.sh'
2020/03/09 01:04:14 packer-builder-lxd plugin: [INFO] RPC endpoint: Communicator ended with: 0
2020/03/09 01:04:14 [INFO] RPC client: Communicator ended with: 0
2020/03/09 01:04:14 [INFO] RPC endpoint: Communicator ended with: 0
2020/03/09 01:04:14 packer-provisioner-shell plugin: [INFO] RPC client: Communicator ended with: 0
2020/03/09 01:04:14 packer-builder-lxd plugin: Executing with lxc exec in container: packer-ubuntu-1804-qw rm -f
2020/03/09 01:04:14 packer-builder-lxd plugin: Executing lxc exec: /bin/sh []string{"/bin/sh", "-c", "lxc exec packer-ubuntu-1804-qw -- /bin/sh -c \"rm -f \""}
2020/03/09 01:04:23 packer-builder-lxd plugin: lxc exec execution exited with '0': 'rm -f '
2020/03/09 01:04:23 packer-builder-lxd plugin: [INFO] RPC endpoint: Communicator ended with: 0
2020/03/09 01:04:23 [INFO] RPC client: Communicator ended with: 0
2020/03/09 01:04:23 [INFO] RPC endpoint: Communicator ended with: 0
2020/03/09 01:04:23 packer-provisioner-shell plugin: [INFO] RPC client: Communicator ended with: 0
2020/03/09 01:04:23 [INFO] (telemetry) ending shell
2020/03/09 01:04:23 [INFO] (telemetry) Starting provisioner ansible
==> ubuntu-1804-qw: Provisioning with Ansible...
2020/03/09 01:04:23 packer-provisioner-ansible plugin: SSH proxy: serving on 127.0.0.1:38617
==> ubuntu-1804-qw: Executing Ansible: ansible-playbook --extra-vars packer_build_name=ubuntu-1804-qw packer_builder_type=lxd -o IdentitiesOnly=yes -i /tmp/packer-provisioner-ansible840290864 /labo/infrabuilder/packer/config.yml -e ansible_ssh_private_key_file=/tmp/ansible-key135327957 -vvvv
    ubuntu-1804-qw: ansible-playbook 2.9.6
    ubuntu-1804-qw:   config file = /etc/ansible/ansible.cfg
    ubuntu-1804-qw:   configured module search path = [u'/home/fred/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
    ubuntu-1804-qw:   ansible python module location = /usr/lib/python2.7/dist-packages/ansible
    ubuntu-1804-qw:   executable location = /usr/bin/ansible-playbook
    ubuntu-1804-qw:   python version = 2.7.17 (default, Nov  7 2019, 10:07:09) [GCC 7.4.0]
    ubuntu-1804-qw: Using /etc/ansible/ansible.cfg as config file
    ubuntu-1804-qw: setting up inventory plugins
    ubuntu-1804-qw: host_list declined parsing /tmp/packer-provisioner-ansible840290864 as it did not pass its verify_file() method
    ubuntu-1804-qw: script declined parsing /tmp/packer-provisioner-ansible840290864 as it did not pass its verify_file() method
    ubuntu-1804-qw: auto declined parsing /tmp/packer-provisioner-ansible840290864 as it did not pass its verify_file() method
    ubuntu-1804-qw: Parsed /tmp/packer-provisioner-ansible840290864 inventory source with ini plugin
    ubuntu-1804-qw: Loading callback plugin default of type stdout, v2.0 from /usr/lib/python2.7/dist-packages/ansible/plugins/callback/default.pyc
    ubuntu-1804-qw:
    ubuntu-1804-qw: PLAYBOOK: config.yml ***********************************************************
    ubuntu-1804-qw: Positional arguments: /labo/infrabuilder/packer/config.yml
    ubuntu-1804-qw: become_method: sudo
    ubuntu-1804-qw: inventory: (u'/tmp/packer-provisioner-ansible840290864',)
    ubuntu-1804-qw: forks: 5
    ubuntu-1804-qw: tags: (u'all',)
    ubuntu-1804-qw: extra_vars: (u'packer_build_name=ubuntu-1804-qw packer_builder_type=lxd -o IdentitiesOnly=yes', u'ansible_ssh_private_key_file=/tmp/ansible-key135327957')
    ubuntu-1804-qw: verbosity: 4
    ubuntu-1804-qw: connection: smart
    ubuntu-1804-qw: timeout: 10
    ubuntu-1804-qw: 1 plays in /labo/infrabuilder/packer/config.yml
    ubuntu-1804-qw:
    ubuntu-1804-qw: PLAY [default] *****************************************************************
    ubuntu-1804-qw: META: ran handlers
    ubuntu-1804-qw:
    ubuntu-1804-qw: TASK [Ajout user ansibleuser avec cle ssh] ******************************************
    ubuntu-1804-qw: task path: /labo/infrabuilder/packer/config.yml:8
    ubuntu-1804-qw: <127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: ansibleuser
    ubuntu-1804-qw: <127.0.0.1> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=38617 -o 'IdentityFile="/tmp/ansible-key135327957"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="ansibleuser"' -o ConnectTimeout=10 -o ControlPath=/home/fred/.ansible/cp/2709ef5cc6 127.0.0.1 '/bin/sh -c '"'"'echo ~ansibleuser && sleep 0'"'"''
2020/03/09 01:04:25 packer-provisioner-ansible plugin: SSH proxy: accepted connection
2020/03/09 01:04:25 packer-provisioner-ansible plugin: authentication attempt from 127.0.0.1:60838 to 127.0.0.1:38617 as ansibleuser using none
2020/03/09 01:04:25 packer-provisioner-ansible plugin: authentication attempt from 127.0.0.1:60838 to 127.0.0.1:38617 as ansibleuser using publickey
2020/03/09 01:04:25 packer-provisioner-ansible plugin: new env request: LANG=en_US.UTF-8
2020/03/09 01:04:25 packer-provisioner-ansible plugin: new exec request: /bin/sh -c 'echo ~ansibleuser && sleep 0'
2020/03/09 01:04:25 packer-builder-lxd plugin: Executing with lxc exec in container: packer-ubuntu-1804-qw /bin/sh -c 'echo ~ansibleuser && sleep 0'
2020/03/09 01:04:25 packer-builder-lxd plugin: Executing lxc exec: /bin/sh []string{"/bin/sh", "-c", "lxc exec packer-ubuntu-1804-qw -- /bin/sh -c \"/bin/sh -c 'echo ~ansibleuser && sleep 0'\""}
```

**Comment by eniac111** [comment]
I can confirm it with the LXD builder and Ansible

**Comment by SwampDragons** [comment]
I think this is a duplicate of https://github.com/hashicorp/packer/issues/7667; as a workaround you can downgrade to ansible version < 2.8, or use the shell-local provisioner to call ansible instead of using the ansible-remote provisioner.

**Comment by quanticware** [comment]
OK, not working  with ansible 2.7.16, 2.7.9 .
And with shell-local I don't know how to pass to lxd container builder.

Hope fix soon.

Thanks
